src/pages/LoginSignIn/LoginSignInForm.js

```diff
@@ -0,0 +1,223 @@
+import React, { Component } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import API_URLS from '../../config';
+import ModalLogoLayout from '../CommonComponents/ModalLogoLayout';
+import './LoginSignInForm.scss';
+
+export default class LoginSignInForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      pw: '',
+      name: '',
+    };
+  }
+
+  requestLogin = e => {
+    e.preventDefault();
+    fetch(API_URLS.SIGNIN, {
+      method: 'POST',
+      body: JSON.stringify({
+        email: this.state.id,
+        password: this.state.pw,
+      }),
+    })
+      .then(res => {
+        if (res.status === 200) {
+          return res.json();
+        }
+      })
+      .then(res => {
+        if (res) {
+          localStorage.setItem('TOKEN', res.token);
+          this.props.history.push('/review');
+        } else {
+          alert('로그인 하세요');
+        }
+      });
+  };
+
+  handleInput = e => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  handleDeleteBtn = e => {
+    e.preventDefault();
+    this.setState({
+      [e.currentTarget.previousSibling.name]: '',
+    });
+  };
+
+  checkIdValid = () => {
+    const { id } = this.state;
+    if (id) {
+      const checkIdCondition =
+        /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+      return checkIdCondition.test(id);
+    } else return true;
+  };
+
+  checkPwValid = () => {
+    const { pw } = this.state;
+    if (pw) {
+      const checkChars = /(?=.*[A-Za-z])/;
+      const checkNums = /(?=.*\d)/;
+      const checkMarks = /(?=.*[$@$!%*#?&])/;
+      const checkCounts = /^[A-Za-z\d$@$!%*#?&]{10,}$/;
+      return (
+        checkCounts.test(pw) &&
+        [checkChars.test(pw), checkNums.test(pw), checkMarks.test(pw)].filter(
+          Boolean
+        ).length >= 2
+      );
+    } else return true;
```

Comment:
이 if else 구문에서는 만약 pw에 값이 없다면(빈스트링) true를 리턴하는 것으로 보이는데 그러면 유효성 검사가 잘 못 되고 있는거 아닐까요??

```suggestion
    if (pw) {
      const checkChars = /(?=.*[A-Za-z])/;
      const checkNums = /(?=.*\d)/;
      const checkMarks = /(?=.*[$@$!%*#?&])/;
      const checkCounts = /^[A-Za-z\d$@$!%*#?&]{10,}$/;
      return (
        checkCounts.test(pw) &&
        [checkChars.test(pw), checkNums.test(pw), checkMarks.test(pw)].filter(
          Boolean
        ).length >= 2
      );
    }
```
이런식으로 처리하면 pw에 값이 있을 경우 유효성 검사를 한 결과를 리턴하고, 만약 pw에 값이 없다면 undefined가 return되어서 자연스럽게 false 처리될 것 같은데
혹시 pw에 값이 없을때 true를 리턴한 이유가 있으신가요??

Reply:
연욱님 요거 그렇게 하면 처음 렌더 되었을 때부터 오류 모드 디자인 떠가지고 넣었던거 같아요 단 그렇게 했을 경우 처음 렌더시에 input 값이 아무것도 없음에도 버튼이 눌려서 문제가 되고 있습니다. 좀 제가 다시 뜯어 고쳐야할거 같아요 😅